### PR TITLE
feat: add stub-mcp runtime for MCP integration testing

### PR DESCRIPTION
**Pull request by nick-pape**
## Summary
- Add a new `stub-mcp` runtime that makes **real MCP tool calls** via `@modelcontextprotocol/sdk`, exercising the full MCP tool-call chain (runtime → MCP HTTP client → MCP server → tool handler → gRPC → SQLite → response back to UI) during E2E tests
- **Bug fix (production):** The WS bridge's `startTaskSession` did not pass `mcpUrl`/`mcpToken` in PowerLine SpawnRequests, so **all runtimes** started from the web UI silently lacked MCP broker access. Only CLI-initiated tasks (gRPC path) had it. This fix adds the same scoped-token generation to the WS path, benefiting claude-code, copilot, codex, and all other runtimes — not just stub-mcp.
- Fix `patchWsForStubRuntime` to use `personaId` instead of the dead `runtime` field (the server ignores `runtime` and resolves it from the persona cascade)
- Add `data-testid` attributes to `ToolResultEvent` for robust E2E selectors
- Add 8 unit tests and 2 E2E integration tests; all 291+ E2E tests pass with zero regressions

## Test plan
- [x] 8 unit tests for StubMcpRuntime (fallback path, lifecycle, kill, fail, resume, MCP success, MCP error)
- [x] E2E: real MCP tool_use and tool_result rendered in UI with correct pairing
- [x] E2E: paired tool_use consumed by tool_result (no standalone card), success indicator shown
- [x] Full E2E suite passes with no regressions

### CODE REVIEW

File: packages/server/src/ws-bridge.ts [hunk 1]
```diff
@@ -340,6 +343,15 @@ async function startTaskSession(
...
 
+  // Build MCP broker URL + scoped token so runtimes can call the MCP server.
+  const mcpPort = parseInt(process.env.GRACKLE_MCP_PORT || String(DEFAULT_MCP_PORT), 10);
+  const mcpDialHost = toDialableHost(process.env.GRACKLE_HOST || "127.0.0.1");
+  const mcpUrl = `http://${mcpDialHost}:${mcpPort}/mcp`;
+  const mcpToken = createScopedToken(
+    { sub: freshTask.id, pid: freshTask.projectId, per: resolved.personaId, sid: sessionId },
+    loadOrCreateApiKey(),
+  );
+
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply packages/server/src/ws-bridge.ts]: Updated the PR description to prominently call out the ws-bridge bug fix as a production behavior change.
